File: src/brand/brand.service.ts
import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
import { InjectModel } from '@nestjs/sequelize';
import { Brand } from './brand.model';
import { CreateBrandDto } from './dto/createBrand.dto';
import { RemoveBrandDto } from './dto/removeBrand.dto';

@Injectable()
export class BrandService {
  constructor(@InjectModel(Brand) private brandRepository: typeof Brand) {}

  async create(dto: CreateBrandDto) {
    console.log(dto);

    const brand = await this.brandRepository.create(dto);

    return brand;
  }

  async getAllBrands() {
    const brands = await this.brandRepository.findAll({
      include: { all: true },
    });

    return brands;
  }

  async removeBrand(dto: RemoveBrandDto) {
    const candidat = await this.getBrandByName(dto.name);

    return candidat;
  }

  async updateBrand(dto: CreateBrandDto) {
    const candidat = await this.getBrandByName(dto.name);

    return candidat;
  }

  async getBrandByName(name: string) {
    const brand = await this.brandRepository.findOne({ where: { name } });

    if (!brand) {
      throw new HttpException(
        'The Brand ' + name + ' does not exist',
        HttpStatus.BAD_REQUEST,
      );
    }

    return brand;
  }
}

File: src/main/java/com/example/demo/brand/BrandService.java
package com.example.demo.brand;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class BrandService {
  private final BrandRepository brandRepository;
  @Autowired
  public BrandService(BrandRepository brandRepository) {
    this.brandRepository = brandRepository;
  }

  public Brand create(Brand brand) {
    return null;
  }

  public List<Brand> getAllBrands() {
    return null;
  }

  public void removeBrand(Long id) {
  }

  public Brand updateBrand(Brand dto) {
    return null;
  }

  public Brand getBrandByName(String name) {
    return null;
  }
}
